plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererLanguageManagerTest.kt

```diff
@@ -70,26 +81,97 @@ class CodeWhispererLanguageManagerTest {
         assertThat(lang1).isSameAs(lang2)
     }
 
+    @Test
+    fun `test getProgrammingLanguage(virtualFile) by fileType`() {
+        testGetProgrammingLanguageUtil<CodeWhispererJava>(fileTypeNames = listOf("java", "Java", "JAVA"), fileExtensions = listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererPython>(listOf("python", "Python"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererJavaScript>(listOf("javascript", "JavaScript"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererJsx>(listOf("jsx harmony"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererTsx>(listOf("typescript jsx"), listOf("tsx"))
+        testGetProgrammingLanguageUtil<CodeWhispererTypeScript>(listOf("typescript", "TypeScript"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererCsharp>(listOf("c#", "C#"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererGo>(listOf("go", "Go"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererKotlin>(listOf("kotlin", "Kotlin"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererPhp>(listOf("php", "Php"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererRuby>(listOf("ruby", "Ruby"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererScala>(listOf("scala", "Scala"), listOf(""))
+        testGetProgrammingLanguageUtil<CodeWhispererSql>(listOf("sql", "Sql"), listOf(""))
+//        testGetProgrammingLanguageUtil<CodeWhispererCpp>(listOf("c++"), listOf(""))
```

Comment:
nit: why these 2 are commented out?

Reply:
cuz the original implementation doesn't have C/Cpp https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt#L56-L77. So C/Cpp are purely determined by file extensions for now.

Reply:
I just checked the `virtualFile.fileType` of `.c` and `.cpp` file and both fileType show `cpp`. So I guess that's the reason why we don't have c and cpp file type condition here.